### PR TITLE
Accept {request_options, [{Key, Value}]} and pass to hackney:request/5

### LINKED ISSUE #80: Possibility to give hackney options
Possibility to give `hackney:request/5` additional options, without having to give a _whole_ function in  to replace `{request, ?DEFAULT_REQUEST_FUN}`.

Currently `?DEFAULT_REQUEST_FUN` is hardcoded to use two specific options when calling `hackney:request/5`.
```
-define(DEFAULT_REQUEST_FUN,        fun katt_callbacks:request/3).
```
In `katt_util.erl`:
```erlang
external_http_request(Url, Method, Hdrs, Body, Timeout, []) ->
  ...
  Options = [ {recv_timeout, Timeout}
            , {insecure, true}
            ],
  case hackney:request(Method, BUrl, BHdrs, Body, Options) of
```

### PR DESCRIPTION
**Pull request by brucify**
Example:
```erl
katt:run("hello.apib", [{request_options, [{ssl_options, {certfile, "path"}, {keyfile, "path"}}]).
```

Fixes #80

Edit:
```erlang
katt:run("hello.apib", [{hackney_options, [{ssl_options, {certfile, "path"}, {keyfile, "path"}}]).
```

### CODE REVIEW

**Comment by andreineculau** [comment]
Cursory review on my phone: looks good. Thanks for the input. At first one might say "but why when you can redefine the request function?" but the default function is far from a oneliner.

Quick thoughts:
- rename to hackney_options for clarity and future proofing
- I believe the request function should accept Params as the last argument instead of Options. This way you can also extract the hackney options only where they are needed e.g. not in a case where you override the request fun and use smth other than hackney
- Maybe make sure that the options passed to hackney don't have duplicate keys when you merge with recv_timeout and insecure. I can't check right now Hackney's behavior.

PS: not sure if it's logged in the git history but this has been implemented before and later removed because i thought it's a nice to have without any real demand. Great that there's demand! :)

**Comment by brucify** [comment]
Thx for the feedback. Fixed in a new commit

File: src/katt_callbacks.erl [hunk 1]
```diff
@@ -263,7 +265,7 @@ http_request( #katt_request{ method = Method
   Hdrs1 = proplists:delete("x-katt-request-sleep", Hdrs0),
   Hdrs = proplists:delete("x-katt-request-timeout", Hdrs1),
   timer:sleep(Sleep),
-  katt_util:external_http_request(Url, Method, Hdrs, Body, Timeout, []).
+  katt_util:external_http_request(Url, Method, Hdrs, Body, Timeout, Params).
```
Comment (andreineculau) [inline src/katt_callbacks.erl:268]: send only the hackney_options as HackneyOpts, not the entire katt params to keep katt_util:external_http_request minimalistic

**Review by andreineculau — CHANGES_REQUESTED** [review]
Sorry for the long turnaround time! Looks neat!
I have one change request though, if possible. See inline.

**Pull request merged**